src/frontend/App.tsx

```diff
@@ -46,16 +46,30 @@ if (applicationId?.endsWith('.dev') || applicationId?.endsWith('.pre')) {
 }
 
 const sentryDebug = applicationId?.endsWith('.dev');
+const appMetricsOptIn = sentryEnvironment !== 'production';
+let navigationIntegration:
+  | ReturnType<(typeof Sentry)['reactNavigationIntegration']>
+  | undefined = undefined;
 const sentryUserId = getSentryUserId({now: new Date(), storage});
 
 Sentry.init({
   dsn: 'https://e0e02907e05dc72a6da64c3483ed88a6@o4507148235702272.ingest.us.sentry.io/4507170965618688',
-  tracesSampleRate: 1.0,
+  tracesSampleRate: appMetricsOptIn ? 1.0 : 0, // Only enable tracing once we have user consent
+  enableUserInteractionTracing: appMetricsOptIn, // Only enable user interaction tracing once we have user consent
   environment: sentryEnvironment,
   debug: sentryDebug, // If `true`, Sentry will try to print out useful debugging information if something goes wrong with sending the event. Set it to `false` in production
   initialScope: {user: {id: sentryUserId}},
 });
 
+if (appMetricsOptIn) {
+  Sentry.setTag('appMetricsOptIn', 'true');
+  navigationIntegration = Sentry.reactNavigationIntegration({
+    enableTimeToInitialDisplay: true,
+    ignoreEmptyBackNavigationTransactions: false,
+  });
+  Sentry.getClient()?.addIntegration(navigationIntegration);
```

Comment:
What is this line for? I don't see it in the [documentation for react navigation setup](https://docs.sentry.io/platforms/react-native/tracing/instrumentation/react-navigation/)